### DISCUSSION TITLE
Custom new paragraph

### DISCUSSION THREAD

**Discussion by af-sko** [ueberdosis/tiptap#1712] [General Questions] [answered]
Hi,

just asking if it is possible to hook into the process of adding a new paragraph.
I just trying to extend the actual handling on adding a new paragraph with some additional, e.g some add some attributes to the html tag.

Is this somehow possible with an extension ?

**Comment by hanspagel** [comment, marked as answer]
Sure! Does that help? https://www.tiptap.dev/guide/styling#extensions

```js
new Editor({
  extensions: [
    Document,
    Paragraph.configure({
      HTMLAttributes: {
        class: 'my-custom-paragraph',
      },
    }),
    Heading.configure({
      HTMLAttributes: {
        class: 'my-custom-heading',
      },
    }),
    Text,
  ],
})
```

**Reply by af-sko** [reply]
Ahh, so the `renderHtml` method of .extend is called also when editing the text. Already used this to add the attribute on adding the content in setup. 🤦 

Thank you :)
